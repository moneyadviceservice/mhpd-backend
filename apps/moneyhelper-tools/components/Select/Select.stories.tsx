import { StoryFn } from '@storybook/react';
import { Select, Props } from './Select';

const StoryProps = {
  title: 'Components/TOOLS/Select',
  component: Select,
  argTypes: {},
};

const Template: StoryFn<Props> = (args) => <Select {...args} />;

export const Default = Template.bind({});
Default.args = {
  name: 'name1',
  options: [{ text: 'Some text', value: 'Some value' }],
};

export const EmptyItemText = Template.bind({});
EmptyItemText.args = {
  name: 'name1',
  emptyItemText: 'Please choose an item',
  options: [{ text: 'Some text', value: 'Some value' }],
};

export const HideEmptyItem = Template.bind({});
HideEmptyItem.args = {
  name: 'name1',
  hideEmptyItem: true,
  options: [{ text: 'Some text', value: 'Some value' }],
};

export const HidePlaceholder = Template.bind({});
HidePlaceholder.args = {
  name: 'name1',
  hidePlaceholder: true,
  options: [{ text: 'Some text', value: 'Some value' }],
};

export default StoryProps;
