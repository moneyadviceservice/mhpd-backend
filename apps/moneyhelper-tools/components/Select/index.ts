export { Select } from './Select';
export type { Props as SelectProps } from './Select';
