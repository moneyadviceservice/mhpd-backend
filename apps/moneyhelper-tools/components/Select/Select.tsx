import { Icon, IconType } from '@maps-react/common/components/Icon';
import { twMerge } from 'tailwind-merge';
import { ChangeEvent, ReactNode, SelectHTMLAttributes } from 'react';
export interface Options {
  text: ReactNode;
  value: string;
}

export type Props = SelectHTMLAttributes<HTMLSelectElement> & {
  emptyItemText?: string;
  hideEmptyItem?: boolean;
  hidePlaceholder?: boolean;
  options: Options[];
  onChange?: (event: ChangeEvent<HTMLSelectElement>) => void;
  className?: string;
  selectClassName?: string;
};

export const Select = ({
  defaultValue = '',
  emptyItemText,
  hideEmptyItem,
  hidePlaceholder,
  options,
  onChange,
  className,
  selectClassName,
  ...props
}: Props) => {
  return (
    <div
      className={twMerge(
        'relative block focus-within:border-purple-600 border-transparent border focus-within:rounded focus-within:shadow-select-focus',
        className,
      )}
    >
      <span className="print:hidden flex absolute pointer-events-none items-center inset-y-0 right-0 w-10 pl-1 pr-1 text-white bg-pink-600 rounded-r shadow-bottom-gray focus:shadow-none">
        <Icon className="h-full" type={IconType.CHEVRON_DOWN}></Icon>
      </span>

      <select
        {...props}
        defaultValue={!props.value ? defaultValue : undefined}
        className={twMerge(
          'print:p-1 text-gray-500 border-gray-400 border-y border-l w-full outline-none block text-md h-10 pl-3 pr-12 bg-white rounded focus:border-white tool-dd',
          selectClassName,
        )}
        onChange={onChange}
      >
        {!hideEmptyItem && (
          <option disabled hidden={hidePlaceholder} value="">
            {emptyItemText}
          </option>
        )}
        {options.map(({ text, value }) => (
          <option key={value} value={value}>
            {text}
          </option>
        ))}
      </select>
    </div>
  );
};
