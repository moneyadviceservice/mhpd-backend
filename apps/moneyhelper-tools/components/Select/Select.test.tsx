import { render } from '@testing-library/react';
import { Select } from './Select';

const options = [
  { text: 'English', value: 'en' },
  { text: 'Spanish', value: 'es' },
];

describe('Select component', () => {
  it('renders correctly', () => {
    const { container } = render(<Select name="language" options={options} />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders with empty item text', () => {
    const { container } = render(
      <Select
        emptyItemText="Select a language"
        name="language"
        options={options}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders with hidden empty item', () => {
    const { container } = render(
      <Select hideEmptyItem name="language" options={options} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders with hidden placeholder', () => {
    const { container } = render(
      <Select hidePlaceholder name="language" options={options} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
