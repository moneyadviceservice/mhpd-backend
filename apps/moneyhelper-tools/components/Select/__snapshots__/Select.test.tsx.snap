// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component renders correctly 1`] = `
<div
  class="relative block focus-within:border-purple-600 border-transparent border focus-within:rounded focus-within:shadow-select-focus"
>
  <span
    class="print:hidden flex absolute pointer-events-none items-center inset-y-0 right-0 w-10 pl-1 pr-1 text-white bg-pink-600 rounded-r shadow-bottom-gray focus:shadow-none"
  >
    <svgr-mock
      classname="h-full"
    />
  </span>
  <select
    class="print:p-1 text-gray-500 border-gray-400 border-y border-l w-full outline-none block text-md h-10 pl-3 pr-12 bg-white rounded focus:border-white tool-dd"
    name="language"
  >
    <option
      disabled=""
      selected=""
      value=""
    />
    <option
      value="en"
    >
      English
    </option>
    <option
      value="es"
    >
      Spanish
    </option>
  </select>
</div>
`;

exports[`Select component renders with empty item text 1`] = `
<div
  class="relative block focus-within:border-purple-600 border-transparent border focus-within:rounded focus-within:shadow-select-focus"
>
  <span
    class="print:hidden flex absolute pointer-events-none items-center inset-y-0 right-0 w-10 pl-1 pr-1 text-white bg-pink-600 rounded-r shadow-bottom-gray focus:shadow-none"
  >
    <svgr-mock
      classname="h-full"
    />
  </span>
  <select
    class="print:p-1 text-gray-500 border-gray-400 border-y border-l w-full outline-none block text-md h-10 pl-3 pr-12 bg-white rounded focus:border-white tool-dd"
    name="language"
  >
    <option
      disabled=""
      selected=""
      value=""
    >
      Select a language
    </option>
    <option
      value="en"
    >
      English
    </option>
    <option
      value="es"
    >
      Spanish
    </option>
  </select>
</div>
`;

exports[`Select component renders with hidden empty item 1`] = `
<div
  class="relative block focus-within:border-purple-600 border-transparent border focus-within:rounded focus-within:shadow-select-focus"
>
  <span
    class="print:hidden flex absolute pointer-events-none items-center inset-y-0 right-0 w-10 pl-1 pr-1 text-white bg-pink-600 rounded-r shadow-bottom-gray focus:shadow-none"
  >
    <svgr-mock
      classname="h-full"
    />
  </span>
  <select
    class="print:p-1 text-gray-500 border-gray-400 border-y border-l w-full outline-none block text-md h-10 pl-3 pr-12 bg-white rounded focus:border-white tool-dd"
    name="language"
  >
    <option
      value="en"
    >
      English
    </option>
    <option
      value="es"
    >
      Spanish
    </option>
  </select>
</div>
`;

exports[`Select component renders with hidden placeholder 1`] = `
<div
  class="relative block focus-within:border-purple-600 border-transparent border focus-within:rounded focus-within:shadow-select-focus"
>
  <span
    class="print:hidden flex absolute pointer-events-none items-center inset-y-0 right-0 w-10 pl-1 pr-1 text-white bg-pink-600 rounded-r shadow-bottom-gray focus:shadow-none"
  >
    <svgr-mock
      classname="h-full"
    />
  </span>
  <select
    class="print:p-1 text-gray-500 border-gray-400 border-y border-l w-full outline-none block text-md h-10 pl-3 pr-12 bg-white rounded focus:border-white tool-dd"
    name="language"
  >
    <option
      disabled=""
      hidden=""
      selected=""
      value=""
    />
    <option
      value="en"
    >
      English
    </option>
    <option
      value="es"
    >
      Spanish
    </option>
  </select>
</div>
`;
